### DISCUSSION TITLE
Storing data fetched from RTK Query into local state for editing

### DISCUSSION THREAD

**Discussion by AsuraKev** [reduxjs/redux-toolkit#3853] [Q&A]
Hi there,

I have a use case where I ll need to fetch some data via useQuery and would need to store the data as a default value on a form which is later on edited by users. Once the user is done with editing, I then use useMutation to commit the change to the server.

Question is how do we correctly store the data from useQuery into local state of a component?

thanks

**Comment by lamhungypl** [comment]
This scenario totally depends on how you deal with the form state.  E.g React-hook-form
https://react-hook-form.com/docs/useform#values
```typescript
function App() {
  const {data: values} = useQuery()
  const [trigger] = useMutation()
 
 
  useForm({
    values, // will get updated once values returns
  })
  
  const onSubmit = (formValues)=>{
  trigger(formValues)
  }
}




```

**Reply by AsuraKev** [reply]
Yep that looks great when using a form solution. What if we are doing manually? How would we store fetched data into state for client modification?
